hasher-matcher-actioner/hmalib/models.py

```diff
@@ -107,3 +131,144 @@ def to_dynamodb_item(self) -> dict:
     def to_sqs_message(self) -> dict:
         # TODO add method for when matches are added to a sqs
         raise NotImplementedError
+
+    @classmethod
+    def get_from_content_key(
+        cls, table: Table, content_key: str
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_content_key(
+            table,
+            cls.get_dynamodb_content_key(content_key),
+            cls.SIGNAL_KEY_PREFIX,
+            cls.SIGNAL_TYPE,
+        )
+        return cls._result_items_to_records(items)
+
+    @classmethod
+    def get_from_signal(
+        cls, table: Table, signal_source: str, signal_id: t.Union[str, int]
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_signal_key(
+            table,
+            cls.get_dynamodb_signal_key(signal_source, signal_id),
+            cls.SIGNAL_TYPE,
+        )
+        return cls._result_items_to_records(items)
+
+    @classmethod
+    def get_from_time_range(
+        cls, table: Table, start_time: str, end_time: str
+    ) -> t.List["PDQMatchRecord"]:
+        items = MatchRecordQuery.from_time_range(
+            table, cls.get_dynamodb_type_key(cls.SIGNAL_TYPE), start_time, end_time
+        )
+        return cls._result_items_to_records(items)
+
+    @staticmethod
+    def _result_items_to_records(items: t.List[t.Dict]) -> t.List["PDQMatchRecord"]:
+        records = []
+        for item in items:
+            records.append(
+                PDQMatchRecord(
+                    item["PK"],
+                    item["ContentHash"],
+                    item["UpdatedAt"],
+                    item["SK"],
+                    item["SignalSource"],
+                    item["SignalHash"],
+                )
+            )
+        return records
```

Comment:
alt, ignorable - This might be a good candidate for a list comprehension

  return [PDQMatchRecord(...) for item in items]